### PR TITLE
Report with aggregation

### PR DESCRIPTION
**Pull request by ZMarouani**
This PR is for creation of reports with grouping .

### CODE REVIEW

File: galite-core/src/main/kotlin/org/kopi/galite/util/LineBreaker.kt [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+            }
+            --i
+          }
+          if (start != 0) {
+            target.append('\n')
+          }
+          if (last != -1) {
+            target.append(line.substring(0, last + 1))
+          }
+          start += col
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/util/LineBreaker.kt:102]: for(start in 0..length step col)

File: galite-core/src/main/kotlin/org/kopi/galite/util/LineBreaker.kt [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+     *
+     * @param        source                the source text with white space
+     * @param        col                the width of the text area
+     */
+    fun modelToText(source: String?, col: Int): String? {
+      return if (source != null) {
+        val target = StringBuffer()
+        val length = source.length
+        var start = 0
+        while (start < length) {
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/util/LineBreaker.kt:86]: for(start in 0..length step col)

File: galite-core/src/main/kotlin/org/kopi/galite/util/LineBreaker.kt [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+    /**
+     * Replaces new-lines by blanks
+     *
+     * @param        source        the source text with carriage return
+     * @param        col        the width of the text
+     * @param        fixed        is it a fixed text ?
+     */
+    fun textToModel(source: String, col: Int, lin: Int, fixed: Boolean): String? {
+      val target = StringBuffer()
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/util/LineBreaker.kt:45]: Use stringBuilder

File: galite-core/src/main/kotlin/org/kopi/galite/util/LineBreaker.kt [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+    fun addBreakForWidth(source: String?, width: Int): String {
+      var source = source
+      val boundary: BreakIterator
+      var start: Int
+      var length: Int
+      val buffer: StringBuffer
+      if (source == null) {
+        source = ""
+      }
```
Comment (hfazai) [inline galite-core/src/main/kotlin/org/kopi/galite/util/LineBreaker.kt:118]: use orEmpty()

File: galite-core/src/main/kotlin/org/kopi/galite/util/LineBreaker.kt (hunk 3 above)
Comment (ZMarouani) [reply galite-core/src/main/kotlin/org/kopi/galite/util/LineBreaker.kt:45]: *BuildString(){} used

File: galite-core/src/main/kotlin/org/kopi/galite/util/LineBreaker.kt (hunk 4 above)
Comment (ZMarouani) [reply galite-core/src/main/kotlin/org/kopi/galite/util/LineBreaker.kt:118]: Fixed

**Pull request closed without merging**